packages/react-native/React/Fabric/Mounting/ComponentViews/Image/RCTImageComponentView.mm

```diff
@@ -168,24 +168,42 @@ - (void)didReceiveImage:(UIImage *)image metadata:(id)metadata fromObserver:(con
   }
 }
 
-- (void)didReceiveProgress:(float)progress fromObserver:(const void *)observer
+- (void)didReceiveProgress:(float)progress loaded:(int64_t)loaded total:(int64_t)total  fromObserver:(const void *)observer
 {
   if (!_eventEmitter) {
     return;
   }
 
-  static_cast<const ImageEventEmitter &>(*_eventEmitter).onProgress(progress);
+  static_cast<const ImageEventEmitter &>(*_eventEmitter).onProgress(progress, loaded, total);
 }
 
-- (void)didReceiveFailureFromObserver:(const void *)observer
+- (void)didReceiveFailure:(NSError *)error fromObserver:(const void *)observer
 {
   _imageView.image = nil;
 
   if (!_eventEmitter) {
     return;
   }
 
-  static_cast<const ImageEventEmitter &>(*_eventEmitter).onError();
+  ImageErrorInfo info;
+
+  if (error) {
+    info.error = [error.localizedDescription UTF8String];
+    id code = error.userInfo[@"httpStatusCode"];
+    if (code){
+      info.responseCode = new int([code intValue]);
+    }
+    id rspHeaders = error.userInfo[@"httpResponseHeaders"];
+    if (rspHeaders) {
+      info.httpResponseHeaders = new std::map<std::string, std::string>();
```

Comment:
C++ types don't get ARC, and we should not use `new` here. Fields should just be stored by value.

Reply:
oh, let me fix this